### DISCUSSION TITLE
[Request] Make the Color component switchable between RGB and HSV or add a second HSV component.

### DISCUSSION THREAD

**Discussion by KarstenTWolf** [FakeFishGames/Barotrauma#4967] [Ideas and suggestions]
A lot of assemblies like to shift across Hues and that's moderately difficult to get 'accurate', having the color component or a second HSV component would greatly simplify this. 

It's not too difficult maths till you get to where the order changes on H' range.

https://en.wikipedia.org/wiki/HSL_and_HSV#Color_conversion_formulae

**Comment by 3e849f2e5c** [comment]
Implemented a checkbox "Use HSV" to the color component that allows it to take HSV input where signal_r is the hue between 0 and 360, signal_g is the saturation between 0 and 1 and signal_b is the value between 0 and 1 and converts it into RGB output
